lib/cookpod/recipes/view_counter.ex

```diff
@@ -11,18 +14,65 @@ defmodule Cookpod.Recipes.ViewCounter do
 
   def increment(recipe_id), do: GenServer.cast(__MODULE__, {:increment, recipe_id})
 
-  def stats(), do: GenServer.call(__MODULE__, :stats)
+  def stats_stream() do
+    Stream.resource(
+      &stream_start_fun/0,
+      &stream_next_fun/1,
+      fn _ -> :ok end
+    )
+  end
 
   # GenServer API
 
-  def init(state), do: {:ok, state}
+  def init(state) do
+    :ets.new(@table_name, [:named_table, :set, :protected, read_concurrency: true])
+    {:ok, state}
+  end
 
-  def handle_cast({:increment, recipe_id}, state) do
-    recipe_stats = Map.get(state, recipe_id, %{id: recipe_id, views: 0})
-    views_count = Map.fetch!(recipe_stats, :views)
-    new_state = Map.put(state, recipe_id, %{recipe_stats | views: views_count + 1})
+  def handle_cast({:increment, recipe_id}, _state) do
+    recipe_stats = get_recipe_stats(recipe_id)
+    new_state = Map.update!(recipe_stats, :views, &(&1 + 1))
+    set_recipe_stats(recipe_id, new_state)
     {:noreply, new_state}
   end
 
   def handle_call(:stats, _from, state), do: {:reply, Map.values(state), state}
+
+  def get_recipe_stats(recipe_id) do
+    case :ets.lookup(@table_name, recipe_id) do
+      [] -> %{id: recipe_id, views: 0}
+      [{_key, stats}] -> stats
+    end
+  end
+
+  defp set_recipe_stats(recipe_id, stats) do
+    :ets.insert(@table_name, {recipe_id, stats})
+  end
+
+  defp stream_start_fun() do
+    match_spec = [{{:"$1", :"$2"}, [], [:"$2"]}]
+    :ets.select(@table_name, match_spec, @stats_batch_size)
+  end
+
+  defp stream_next_fun(acc) do
+    case acc do
+      @end_of_table ->
+        {:halt, acc}
+
+      {list, @end_of_table} ->
+        case list do
+          [] -> {:halt, acc}
+          _ -> {list, {[], @end_of_table}}
+        end
+
+      {[], continuation} ->
+        case :ets.select(continuation) do
+          {list, next_continuation} -> {list, {[], next_continuation}}
+          _ -> {:halt, acc}
+        end
+
+      {list, continuation} ->
+        {list, {[], continuation}}
+    end
+  end
```

Comment:
По идее должно быть вего 2 клозы. Я думаю, вот так должно сработать

```suggestion
  defp stream_next_fun(acc) do
    case acc do
      @end_of_table ->
        {:halt, acc}

      {list, continuation} ->
        {list, :ets.select(continuation)}
  end
```